### PR TITLE
Update guestbook example to use replication controller instead of a naked pod.

### PR DESCRIPTION
**Pull request by rsokolowski**
Bunch of changes in addition and also to make the change work:
- Added redis-master-controller.json that replaces redis-master.json.
- Removed HostPort from resources in the json files.
- Expanded README.md to include 6th step that sets up service for frontend.
- Reexecuted commands from the example and pasted up to date output.
- Bunch of changes in test files that were relying on redis-master.json file that generated pod resource. I've updated the relevant calls to use replication controller resource.

### CODE REVIEW

**Comment by bgrant0607** [comment]
Tests failed with:
```
F0210 00:53:56.836262   14857 create.go:79] the path "examples/guestbook/redis-master.json" does not exist
```

**Comment by bgrant0607** [comment]
Thanks @rsokolowski! @ddysher could you take a look?

**Comment by ddysher** [comment]
Sure, I'll take a look.

File: examples/guestbook/README.md [hunk 1]
```diff
@@ -46,7 +54,7 @@ Use the file `examples/guestbook/redis-master.json` which describes a single pod
 Create the redis pod in your Kubernetes cluster by running:
 
 ```shell
-$ cluster/kubectl.sh create -f examples/guestbook/redis-master.json
+$ cluster/kubectl.sh create -f examples/guestbook/redis-master-controller.json
 ```
```
Comment (ddysher) [inline examples/guestbook/README.md]: Also show that using `cluster/kubectl.sh get rc` will list the replication controller.

File: examples/guestbook/README.md [hunk 2]
```diff
@@ -323,19 +330,54 @@ if (isset($_GET['cmd']) === true) {
...
+frontend                name=frontend                             name=frontend                10.0.93.211         80
+redis-master            name=redis-master                         name=redis-master            10.0.246.242        6379
+redisslave              name=redis-slave                          name=redis-slave             10.0.72.62          6379
+```
+
+You may need to open the firewall for port 80 using the [console][cloud-console] or the `gcloud` tool. The following command will allow traffic from any source to instances tagged `kubernetes-minion`:
 
 ```shell
-$ gcloud compute instances list
+$ gcloud compute firewall-rules create --allow=tcp:80 --target-tags=kubernetes-minion kubernetes-minion-80
```
Comment (ddysher) [inline examples/guestbook/README.md]: Fix the service with port 8000, instead of requiring 80?  We'd better not use priviledge port.

File: pkg/kubectl/cmd/cmd_test.go [hunk 3]
```diff
@@ -204,43 +203,3 @@ func TestClientVersions(t *testing.T) {
 		}
 	}
 }
-
-func ExamplePrintReplicationController() {
```
Comment (ddysher) [inline pkg/kubectl/cmd/cmd_test.go]: Why is this removed?  Dead code?

**Comment by ddysher** [comment]
@rsokolowski Thanks!  

I haven't looked at kubectl unittest yet.  But I think you'll also need to fix guestbook.sh in e2e-suite.  Can you fix it, rebase/squash?

File: pkg/kubectl/cmd/cmd_test.go (hunk 3 above)
Comment (rsokolowski) [inline pkg/kubectl/cmd/cmd_test.go]: Yes, dead code.

File: examples/guestbook/README.md (hunk 1 above)
Comment (rsokolowski) [inline examples/guestbook/README.md]: Done.

File: examples/guestbook/README.md (hunk 2 above)
Comment (rsokolowski) [inline examples/guestbook/README.md]: Done.

**Comment by rsokolowski** [comment]
Thanks for comments @ddysher. PTAL.

File: examples/limitrange/valid-pod.json [hunk 4]
```diff
@@ -1,20 +1,19 @@
 {
   "id": "valid-pod",
   "kind": "Pod",
-  "apiVersion":"v1beta2",
```
Comment (ddysher) [inline examples/limitrange/valid-pod.json]: why change to v1beta2?

File: examples/limitrange/valid-pod.json [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
   "desiredState": {
     "manifest": {
       "version": "v1beta1",
-      "id": "invalid-pod",
+      "id": "valid-pod",
       "containers": [{
         "name": "kubernetes-serve-hostname",
         "image": "kubernetes/serve_hostname",        
         "cpu": 1000,
-        "memory": 1048576,
```
Comment (ddysher) [inline examples/limitrange/valid-pod.json]: memory should be a valid part of pod, why do you remove it?

**Comment by ddysher** [comment]
Looks good, some minor comments.  Need a rebase.

File: examples/guestbook/README.md [hunk 6]
```diff
@@ -46,30 +54,35 @@ Use the file `examples/guestbook/redis-master.json` which describes a single pod
...
+$ cluster/kubectl.sh get pods
 ```
 
 You'll see a single redis master pod. It will also display the machine that the pod is running on once it gets placed (may take up to thirty seconds):
 
 ```
-NAME                IMAGE(S)            HOST                                                       LABELS              STATUS
-redis-master        dockerfile/redis    kubernetes-minion-2.c.myproject.internal/130.211.156.189   name=redis-master   Running
+POD                                          IP                  CONTAINER(S)            IMAGE(S)                                 HOST                                                           LABELS                                                     STATUS                                                                                                                               
+redis-master-controller-gb50a                10.244.3.7          redis-master            dockerfile/redis                         kubernetes-minion-4.c.hazel-mote-834.internal/104.154.54.203   app=redis,name=redis-master                                Running
```
Comment (zmerlynn) [inline examples/guestbook/README.md]: It's actually because the minions are now created as part of Managed Instance Groups members of which just have unique IDs versus counts.

Comment (ddysher) [inline examples/guestbook/README.md]: Ah, really, I thought it's follow up work for this #3789

File: examples/limitrange/valid-pod.json (hunk 4 above)
Comment (zmerlynn) [inline examples/limitrange/valid-pod.json]: Should we actually be updating to v1beta3 if we're updating at all? cc @roberthbailey

Comment (roberthbailey) [inline examples/limitrange/valid-pod.json]: I wanted to switch examples to v1beta3 but it doesn't look like it's enabled by default yet.

File: examples/guestbook/README.md (hunk 6 above)
Comment (rsokolowski) [inline examples/guestbook/README.md]: Updated.

File: examples/limitrange/valid-pod.json (hunk 4 above)
Comment (rsokolowski) [inline examples/limitrange/valid-pod.json]: Reverted to v1beta2.

File: examples/limitrange/valid-pod.json (hunk 5 above)
Comment (rsokolowski) [inline examples/limitrange/valid-pod.json]: Reverted memory removal.

**Comment by rsokolowski** [comment]
Thanks for comments @ddysher . Applied them and rebased/squashed. I also fixed hack/test-cmd.sh which was broken.

File: pkg/kubectl/cmd/cmd_test.go (hunk 3 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/cmd_test.go]: I don't think this is dead code, it's just testing printer output.

**Comment by rsokolowski** [comment]
Ping, I again synced to head.

File: pkg/kubectl/cmd/cmd_test.go (hunk 3 above)
Comment (zmerlynn) [inline pkg/kubectl/cmd/cmd_test.go]: I don't think this comment was addressed in the last push.

Comment (ddysher) [inline pkg/kubectl/cmd/cmd_test.go]: Prepend Test for the method?

**Comment by ddysher** [comment]
Have you ran e2e test?  Just for precaution even though this is an example update.  I believe a some places in e2e touches the directory.   

You seem to touch very hot files.

**Comment by rsokolowski** [comment]
The relevant e2e tests are passing. There is not that much dependency on examples/guestbook: mainly test-cmd.sh and guestbook.sh

**Comment by ddysher** [comment]
Have you commented on this?
https://github.com/GoogleCloudPlatform/kubernetes/pull/4267#discussion_r24452889

File: pkg/kubectl/cmd/cmd_test.go (hunk 3 above)
Comment (rsokolowski) [inline pkg/kubectl/cmd/cmd_test.go]: For the sake of making progress on the PR (and also because the change was unrelated) I've reverted change in this file.
The code is dead since the function is not called anywhere from the repository, plus this is not a test because it is missing Test prefix and doesn't take *testing.Test as an argument (http://golang.org/pkg/testing/). Even if such conditions did hold it still wouldn't test anything (except that the executed code doesn't crash) since it just prints created replication controller to Stdout and doesn't verify the output.

**Comment by rsokolowski** [comment]
Oops, I've missed the comment - replied.

File: pkg/kubectl/cmd/cmd_test.go (hunk 3 above)
Comment (zmerlynn) [inline pkg/kubectl/cmd/cmd_test.go]: Ah, I didn't notice that either. If you'd like to add a separate commit
that nukes it (since it's unrelated), be my guest. Or one that turns it
into a real test.

**Comment by zmerlynn** [comment]
This LGTM, but deferring to @ddysher for final.

(I was mainly butting my head in here because it was referenced by #4588.)

**Comment by ddysher** [comment]
LGTM, will merge after final rebase.  N.B. can you put a comment on the conflict part so I don't have to go through the whole commit, thx.

File: hack/test-cmd.sh [hunk 7]
```diff
@@ -132,67 +132,68 @@ for version in "${kube_api_versions[@]}"; do
...
+  
+  kubectl delete pods -l name=valid-pod "${kube_flags[@]}"
+  [ ! $(kubectl get pods "${kube_flags[@]}" -lname=valid-pod | grep -q 'valid-pod') ]
+  kubectl create -f examples/limitrange/valid-pod.json "${kube_flags[@]}"
+  kubectl get pods "${kube_flags[@]}" -lname=valid-pod | grep -q 'valid-pod'
   [ ! $(kubectl delete pods "${kube_flags[@]}" ) ]
-  kubectl get pods "${kube_flags[@]}" -lname=redis-master | grep -q 'redis-master'
-  [ ! $(kubectl delete pods --all pods -l name=redis-master "${kube_flags[@]}" ) ]   # not --all and label selector together
+  kubectl get pods "${kube_flags[@]}" -lname=valid-pod | grep -q 'valid-pod'
+  [ ! $(kubectl delete pods --all pods -l name=valid-pod "${kube_flags[@]}" ) ]   # not --all and label selector together
```
Comment (rsokolowski) [inline hack/test-cmd.sh]: Latest conflict was here due to https://github.com/GoogleCloudPlatform/kubernetes/commit/29aae753299981140550397105a3b61c0ea6d38e

**Comment by rsokolowski** [comment]
Thanks for review @ddysher . Rebased, resolved conflict and added comment as requested.
